ongoing_projects/getOngoingProjects.py

```diff
@@ -0,0 +1,146 @@
+import requests
+import datetime
+import json
+import time
+import logging
+import os  # Import the os module
+from dotenv import load_dotenv  # Import load_dotenv
+
+# Load environment variables from .env file
+load_dotenv()
+
+# --- Constants ---
+API_URL = "https://swecris-api.vr.se/v1/projects"
+# Get the token from environment variable
+# To run locally, create a .env file in the project root
+# and add the line: SWECRIS_API_TOKEN="your_actual_token"
+AUTH_TOKEN = os.environ.get("SWECRIS_API_TOKEN")
+# Add a check to ensure the token was loaded
+if not AUTH_TOKEN:
+    raise ValueError("Error: SWECRIS_API_TOKEN environment variable not set or .env file not found.")
+
+PAGE_SIZE = 100
+TODAY = datetime.date.today()
+UNKNOWN = "Unknown"
+
+# --- Topic keywords --- avoid acronymn
+TOPIC_KEYWORDS = {
+    "Antibiotic resistance": ["antibiotic resistance", "antimicrobial resistance", "antimicrobial"],
+    "COVID-19": ["covid","SARS-CoV-2"],
+    "Enteric viruses": ["norovirus", "rotavirus", "enteric virus"],
+    "Infectious diseases": ["infectious disease", "pathogen"],
+    "Influenza": ["influenza", "influensa", "H1N1", "H3N2"],
+    "Mpox": ["mpox", "monkeypox"],
+    "Respiratory Syncytial Virus (RSV)": ["respiratory syncytial virus"]
+}
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+
+# --- Functions ---
+def fetch_all_projects():
+    """Fetch all projects from the SweCRIS API."""
+    logging.info("Fetching projects from SweCRIS...")
+    try:
+        response = requests.get(
+            API_URL,
+            headers={
+                "accept": "application/json",
+                "Authorization": f"Bearer {AUTH_TOKEN}"
+            }
+        )
+        response.raise_for_status()
+    except requests.RequestException as e:
+        logging.error(f"Failed to fetch projects: {e}")
+        return []
+
+    projects = response.json()
+    logging.info(f"Fetched {len(projects)} projects.")
+    return projects
+
+
+def is_ongoing(project):
+    """Check if a project is ongoing based on its start and end dates."""
+    try:
+        start = datetime.datetime.strptime(project["projectStartDate"].split()[0], "%Y-%m-%d").date()
+        end = datetime.datetime.strptime(project["projectEndDate"].split()[0], "%Y-%m-%d").date()
+        return start <= TODAY <= end
+    except (KeyError, ValueError) as e:
+        logging.warning(f"Error parsing dates for project {project.get('projectId', UNKNOWN)}: {e}")
+        return False
+
+
+def match_topics(project):
+    """Match project topics based on keywords."""
+    text = " ".join([
+        str(project.get("projectTitleEn") or project.get("projectTitleSv", "")),
+        str(project.get("projectAbstractEn", ""))
+    ]).lower()
+
+    matched_topics = []
+    for topic, keywords in TOPIC_KEYWORDS.items():
+        if any(keyword.lower() in text for keyword in keywords):
```

Comment:
Using regular expression is more pragmatic as we wanna match words, this might also match a substring.